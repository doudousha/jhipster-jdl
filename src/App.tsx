import React from "react";

import Sidebar from "./components/sidebar/Sidebar";
import Header from "./components/Header";
import Studio from "./components/studio/Studio";
import "./App.scss";

function App() {
  return (
    <div className="wrap app">
      <Header />
      {/* <!-- Code and image --> */}
      <Studio />
      {/* <!-- sidebar --> */}
      <Sidebar />
    </div>
  );
}

export default App;
