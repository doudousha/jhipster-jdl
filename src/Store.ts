import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { studio, StudioState } from "./components/studio/StudioReducer";
import { jhOnline, JhOnlineState } from "./components/JhOnlineReducer";

export interface IRootState {
  readonly studio: StudioState;
  readonly jhOnline: JhOnlineState;
}

export const rootReducer = combineReducers<IRootState>({
  studio,
  jhOnline,
});

export const middlewares = applyMiddleware(thunk);

export const store = createStore(rootReducer, middlewares);
