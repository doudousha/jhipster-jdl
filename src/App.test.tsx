import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import App from "./App";
import { store } from "./Store";

import * as jsdom from "jsdom";
const doc = new jsdom.JSDOM("<!doctype html><html><body></body></html>");
global.document = doc;
global.window = doc.defaultView;
//@ts-ignore
global.document.body.createTextRange = function () {
  return {
    setEnd: function () {},
    setStart: function () {},
    getBoundingClientRect: function () {
      return { right: 0 };
    },
    getClientRects: function () {
      return {
        length: 0,
        left: 0,
        right: 0,
      };
    },
  };
};

test("renders about page text", () => {
  const { getByText } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  const linkElement = getByText(/Hello, this is JDL-Studio/i);
  expect(linkElement).toBeInTheDocument();
});

test("renders default snapshot", () => {
  const comp = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(comp).toMatchSnapshot();
});
