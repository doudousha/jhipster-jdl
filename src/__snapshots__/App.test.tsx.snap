// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="popup-root"
    >
      <div
        class="popup-overlay dark-theme-overlay"
        data-popup="modal"
        data-testid="overlay"
        style="position: fixed; top: 0px; bottom: 0px; left: 0px; right: 0px; display: flex; z-index: 999; pointer-events: auto;"
        tabindex="-1"
      >
        <div
          class="popup-content dark-theme-content"
          id="popup-8"
          role="dialog"
          style="position: relative; margin: auto; pointer-events: auto;"
        >
          <div
            class="dark-theme modal"
          >
            <div
              class="header"
            >
              Warning!
            </div>
            <div
              class="content"
            >
              <p>
                You are using an experimental beta version of JDL Studio. Please use the stable version at
                 
                <a
                  href="https://start.jhipster.tech/jdl-studio/"
                >
                  https://start.jhipster.tech/jdl-studio/
                </a>
                 
                for more features and persistent storage
              </p>
            </div>
            <div
              class="actions"
            >
              <button
                class="button"
              >
                It's OK
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="wrap app"
      >
        <header
          class="dark-theme"
        >
          <div
            class="tools left"
          >
            <a
              class="logo-img"
              href="https://www.jhipster.tech/"
              rel="noopener noreferrer"
              target="_blank"
              title="JHipster website"
            >
              <img
                alt="JHipster logo"
                class="jhi-logo"
                src="logo-jhipster.svg"
              />
            </a>
            <a
              class="logo link"
              title="About JDL-Studio"
            >
              JDL-Studio
            </a>
          </div>
          <div
            class="tools right"
          >
            <div
              class="tools-inner"
            >
              <a
                class="link"
                title="Toggle theme"
              >
                <i
                  class="lni lni-sun size-md "
                />
              </a>
              <a
                class="link"
                title="Cycle graph ranker strategy [current: tight-tree]"
              >
                <i
                  class="lni lni-grid-alt size-md "
                />
              </a>
              <a
                class="link"
                title="Cycle graph direction [current: down]"
              >
                <i
                  class="lni lni-arrow-right size-md "
                />
              </a>
              <span
                class="seperator"
              >
                <i>
                  |
                </i>
              </span>
              <select
                class="template-select"
              >
                <option
                  value=""
                >
                  &lt;Select template&gt;
                </option>
                <option
                  value="default sample"
                >
                  default sample
                </option>
                <option
                  value="simple monolith"
                >
                  simple monolith
                </option>
                <option
                  value="simple microservice"
                >
                  simple microservice
                </option>
              </select>
            </div>
            <div
              class="tools-inner"
            >
              <a
                class="special"
                href="https://start.jhipster.tech/jdl-studio/"
                id="signin"
                title="Go to stable JDL Studio with more features"
              >
                Go to stable JDL Studio
              </a>
              <span
                class="seperator"
              >
                <i>
                  |
                </i>
              </span>
              <a
                class="link"
                download="app-jdl.png"
                id="savebutton"
                title="Download snapshot of this diagram"
              >
                <i
                  class="lni lni-camera size-md "
                />
              </a>
              <a
                class="link"
                download="app.jdl"
                id="saveTextbutton"
                title="Download this JDL source"
              >
                <i
                  class="lni lni-download size-md "
                />
              </a>
              <a
                class="upload-dialog link"
                id="uploadbutton"
                title="Import text file of a JDL"
              >
                <i
                  class="lni lni-upload size-md "
                />
              </a>
              <a
                class="link"
                title="About JDL-Studio"
              >
                <i
                  class="lni lni-question-circle size-md "
                />
              </a>
              <a
                class="link"
                title="Language reference"
              >
                <i
                  class="lni lni-book size-md "
                />
              </a>
              <span
                class="tooltip"
              />
            </div>
          </div>
        </header>
        <div
          class="studio dark-theme "
        >
          <div
            class="line-numbers error"
          />
          <div
            class="line-marker"
          />
          <div
            class="react-codemirror2 code-mirror-editor"
          >
            <div
              class="CodeMirror cm-s-solarized cm-s-dark"
            >
              <div
                style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 0px;"
              >
                <textarea
                  autocapitalize="off"
                  autocorrect="off"
                  spellcheck="false"
                  style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"
                  tabindex="0"
                />
              </div>
              <div
                class="CodeMirror-vscrollbar"
                cm-not-content="true"
                style="display: block; bottom: 0px;"
                tabindex="-1"
              >
                <div
                  style="min-width: 1px; height: 166px;"
                />
              </div>
              <div
                class="CodeMirror-hscrollbar"
                cm-not-content="true"
                style="display: block; right: 0px; left: 0px;"
                tabindex="-1"
              >
                <div
                  style="height: 100%; min-height: 1px; width: 53px;"
                />
              </div>
              <div
                class="CodeMirror-scrollbar-filler"
                cm-not-content="true"
              />
              <div
                class="CodeMirror-gutter-filler"
                cm-not-content="true"
              />
              <div
                class="CodeMirror-scroll"
                tabindex="-1"
              >
                <div
                  class="CodeMirror-sizer"
                  style="margin-left: 0px; min-width: 3px; padding-right: 0px; padding-bottom: 0px;"
                >
                  <div
                    style="position: relative;"
                  >
                    <div
                      class="CodeMirror-lines"
                      role="presentation"
                    >
                      <div
                        role="presentation"
                        style="position: relative; outline: none;"
                      >
                        <div
                          class="CodeMirror-measure"
                        >
                          <span>
                            <span>
                              ​
                            </span>
                            x
                          </span>
                        </div>
                        <div
                          class="CodeMirror-measure"
                        >
                          <pre
                            class="CodeMirror-line"
                            role="presentation"
                          >
                            <span
                              role="presentation"
                              style="padding-right: .1px;"
                            >
                              <span
                                cm-text=""
                                style="display: inline-block; width: 1px; margin-right: -1px;"
                              >
                                 
                              </span>
                            </span>
                          </pre>
                        </div>
                        <div
                          style="position: relative; z-index: 1;"
                        />
                        <div
                          class="CodeMirror-cursors"
                        />
                        <div
                          class="CodeMirror-code"
                          role="presentation"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  style="position: absolute; height: 50px; width: 1px; border-bottom: 0px solid transparent;"
                />
                <div
                  class="CodeMirror-gutters"
                >
                  <div
                    class="CodeMirror-gutter CodeMirror-linenumbers"
                    style="width: 1px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <canvas
            id="canvas"
          />
          <span
            class="error-tooltip"
          >
            Cannot set property 'font' of null
          </span>
          <div
            class="canvas-tools"
            id="canvas-tools"
          >
            <a
              class="link"
              title="Zoom in"
            >
              <i
                class="lni lni-zoom-in size-md "
              />
            </a>
            <a
              class="link"
              title="Reset viewport"
            >
              <i
                class="lni lni-frame-expand size-md "
              />
            </a>
            <a
              class="link"
              title="Zoom out"
            >
              <i
                class="lni lni-zoom-out size-md "
              />
            </a>
          </div>
          <div
            class="canvas-panner"
          />
        </div>
        <div
          class="sidebar  dark-theme"
          id="sidebar"
        >
          <div
            class="content"
          >
            <h3>
              About
            </h3>
            <p>
              Hello, this is JDL-Studio, a tool for drawing
               
              <a
                href="http://www.jhipster.tech/"
                rel="noopener noreferrer"
                target="_blank"
              >
                JHipster
              </a>
               
              JDL diagrams based on the
               
              <a
                href="http://www.jhipster.tech//jdl"
                rel="noopener noreferrer"
                target="_blank"
              >
                JDL syntax
              </a>
              .
               
            </p>
            <p>
              Try and edit the code on the left and watch the diagram change. Any changes are saved to the browser's 
              <i>
                localStorage
              </i>
              , so your diagram should be here the next time, (but no guarantees). If you want persistent storage, sign in with your JHipster Online account to save the JDL.
            </p>
            <p>
              Created by
               
              <a
                href="https://deepu.tech"
                rel="noopener noreferrer"
                target="_blank"
              >
                Deepu KS
              </a>
               
              and hosted on
               
              <a
                href="https://github.com/jhipster/jdl-studio"
                rel="noopener noreferrer"
                target="_blank"
              >
                GitHub
              </a>
              .
            </p>
            <p>
              JDL-Studio was made possible by the cool project
               
              <a
                href="https://github.com/skanaar/nomnoml"
                rel="noopener noreferrer"
                target="_blank"
              >
                nomnoml
              </a>
               
              by
               
              <a
                href="https://github.com/skanaar"
                rel="noopener noreferrer"
                target="_blank"
              >
                Daniel Kallin
              </a>
               
              .
            </p>
            <h2>
              Interaction
            </h2>
            <hr />
            <p>
              The canvas can be panned and zoomed by dragging and scrolling in the right hand third of the canvas.
            </p>
            <img
              alt="interaction tutorial"
              class="invert"
              src="interaction-tutorial.png"
            />
            <p>
              Downloaded image files will be given the filename in the
               
              <kbd>
                #title
              </kbd>
               directive.
            </p>
            <p>
              The editor supports a subset of 
              <i>
                Sublime Text's
              </i>
               
              <a
                href="https://codemirror.net/demo/sublime.html"
                rel="noopener noreferrer"
                target="_blank"
              >
                keymappings
              </a>
              . Hit 
              <code>
                Ctrl-Space
              </code>
               for auto suggestions.
            </p>
            <h2>
              Examples
            </h2>
            <hr />
            <p>
              Refer
               
              <a
                href="http://www.jhipster.tech/jdl"
                rel="noopener noreferrer"
                target="_blank"
              >
                JDL
              </a>
               
              documentation for examples.
            </p>
            <h2>
              Usage
            </h2>
            <hr />
            <p>
              Refer
               
              <a
                href="http://www.jhipster.tech/jdl"
                rel="noopener noreferrer"
                target="_blank"
              >
                JDL
              </a>
               
              documentation for more details.
            </p>
            <label>
              Set font
            </label>
            <code>
              #font: Times
            </code>
            <label>
              Layout the diagram really tight
            </label>
            <code>
              #fontSize: 8
            </code>
            <br />
            <code>
              #spacing: 12
            </code>
            <br />
            <code>
              #padding: 3
            </code>
            <h2>
              Directives
            </h2>
            <hr />
            <p>
              The rendered image style can be customized by setting these directives along with the JDL code. There should not be any empty space before a directive.
            </p>
            <code>
              #arrowSize: 1
            </code>
            <br />
            <code>
              #bendSize: 0.3
            </code>
            <br />
            <code>
              #direction: down | right
            </code>
            <br />
            <code>
              #gutter: 5
            </code>
            <br />
            <code>
              #edgeMargin: 0
            </code>
            <br />
            <code>
              #edges: hard | rounded
            </code>
            <br />
            <code>
              #background: transparent
            </code>
            <br />
            <code>
              #fill: #eee8d5; #fdf6e3
            </code>
            <br />
            <code>
              #fillArrows: false
            </code>
            <br />
            <code>
              #font: Calibri
            </code>
            <br />
            <code>
              #fontSize: 12
            </code>
            <br />
            <code>
              #leading: 1.25
            </code>
            <br />
            <code>
              #lineWidth: 3
            </code>
            <br />
            <code>
              #padding: 8
            </code>
            <br />
            <code>
              #spacing: 40
            </code>
            <br />
            <code>
              #stroke: #33322E
            </code>
            <br />
            <code>
              #title: filename
            </code>
            <br />
            <code>
              #zoom: 1
            </code>
            <br />
            <code>
              #acyclicer: greedy
            </code>
            <br />
            <code>
              #ranker: network-simplex | tight-tree | longest-path
            </code>
            <br />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrap app"
    >
      <header
        class="dark-theme"
      >
        <div
          class="tools left"
        >
          <a
            class="logo-img"
            href="https://www.jhipster.tech/"
            rel="noopener noreferrer"
            target="_blank"
            title="JHipster website"
          >
            <img
              alt="JHipster logo"
              class="jhi-logo"
              src="logo-jhipster.svg"
            />
          </a>
          <a
            class="logo link"
            title="About JDL-Studio"
          >
            JDL-Studio
          </a>
        </div>
        <div
          class="tools right"
        >
          <div
            class="tools-inner"
          >
            <a
              class="link"
              title="Toggle theme"
            >
              <i
                class="lni lni-sun size-md "
              />
            </a>
            <a
              class="link"
              title="Cycle graph ranker strategy [current: tight-tree]"
            >
              <i
                class="lni lni-grid-alt size-md "
              />
            </a>
            <a
              class="link"
              title="Cycle graph direction [current: down]"
            >
              <i
                class="lni lni-arrow-right size-md "
              />
            </a>
            <span
              class="seperator"
            >
              <i>
                |
              </i>
            </span>
            <select
              class="template-select"
            >
              <option
                value=""
              >
                &lt;Select template&gt;
              </option>
              <option
                value="default sample"
              >
                default sample
              </option>
              <option
                value="simple monolith"
              >
                simple monolith
              </option>
              <option
                value="simple microservice"
              >
                simple microservice
              </option>
            </select>
          </div>
          <div
            class="tools-inner"
          >
            <a
              class="special"
              href="https://start.jhipster.tech/jdl-studio/"
              id="signin"
              title="Go to stable JDL Studio with more features"
            >
              Go to stable JDL Studio
            </a>
            <span
              class="seperator"
            >
              <i>
                |
              </i>
            </span>
            <a
              class="link"
              download="app-jdl.png"
              id="savebutton"
              title="Download snapshot of this diagram"
            >
              <i
                class="lni lni-camera size-md "
              />
            </a>
            <a
              class="link"
              download="app.jdl"
              id="saveTextbutton"
              title="Download this JDL source"
            >
              <i
                class="lni lni-download size-md "
              />
            </a>
            <a
              class="upload-dialog link"
              id="uploadbutton"
              title="Import text file of a JDL"
            >
              <i
                class="lni lni-upload size-md "
              />
            </a>
            <a
              class="link"
              title="About JDL-Studio"
            >
              <i
                class="lni lni-question-circle size-md "
              />
            </a>
            <a
              class="link"
              title="Language reference"
            >
              <i
                class="lni lni-book size-md "
              />
            </a>
            <span
              class="tooltip"
            />
          </div>
        </div>
      </header>
      <div
        class="studio dark-theme "
      >
        <div
          class="line-numbers error"
        />
        <div
          class="line-marker"
        />
        <div
          class="react-codemirror2 code-mirror-editor"
        >
          <div
            class="CodeMirror cm-s-solarized cm-s-dark"
          >
            <div
              style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 0px;"
            >
              <textarea
                autocapitalize="off"
                autocorrect="off"
                spellcheck="false"
                style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"
                tabindex="0"
              />
            </div>
            <div
              class="CodeMirror-vscrollbar"
              cm-not-content="true"
              style="display: block; bottom: 0px;"
              tabindex="-1"
            >
              <div
                style="min-width: 1px; height: 166px;"
              />
            </div>
            <div
              class="CodeMirror-hscrollbar"
              cm-not-content="true"
              style="display: block; right: 0px; left: 0px;"
              tabindex="-1"
            >
              <div
                style="height: 100%; min-height: 1px; width: 53px;"
              />
            </div>
            <div
              class="CodeMirror-scrollbar-filler"
              cm-not-content="true"
            />
            <div
              class="CodeMirror-gutter-filler"
              cm-not-content="true"
            />
            <div
              class="CodeMirror-scroll"
              tabindex="-1"
            >
              <div
                class="CodeMirror-sizer"
                style="margin-left: 0px; min-width: 3px; padding-right: 0px; padding-bottom: 0px;"
              >
                <div
                  style="position: relative;"
                >
                  <div
                    class="CodeMirror-lines"
                    role="presentation"
                  >
                    <div
                      role="presentation"
                      style="position: relative; outline: none;"
                    >
                      <div
                        class="CodeMirror-measure"
                      >
                        <span>
                          <span>
                            ​
                          </span>
                          x
                        </span>
                      </div>
                      <div
                        class="CodeMirror-measure"
                      >
                        <pre
                          class="CodeMirror-line"
                          role="presentation"
                        >
                          <span
                            role="presentation"
                            style="padding-right: .1px;"
                          >
                            <span
                              cm-text=""
                              style="display: inline-block; width: 1px; margin-right: -1px;"
                            >
                               
                            </span>
                          </span>
                        </pre>
                      </div>
                      <div
                        style="position: relative; z-index: 1;"
                      />
                      <div
                        class="CodeMirror-cursors"
                      />
                      <div
                        class="CodeMirror-code"
                        role="presentation"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                style="position: absolute; height: 50px; width: 1px; border-bottom: 0px solid transparent;"
              />
              <div
                class="CodeMirror-gutters"
              >
                <div
                  class="CodeMirror-gutter CodeMirror-linenumbers"
                  style="width: 1px;"
                />
              </div>
            </div>
          </div>
        </div>
        <canvas
          id="canvas"
        />
        <span
          class="error-tooltip"
        >
          Cannot set property 'font' of null
        </span>
        <div
          class="canvas-tools"
          id="canvas-tools"
        >
          <a
            class="link"
            title="Zoom in"
          >
            <i
              class="lni lni-zoom-in size-md "
            />
          </a>
          <a
            class="link"
            title="Reset viewport"
          >
            <i
              class="lni lni-frame-expand size-md "
            />
          </a>
          <a
            class="link"
            title="Zoom out"
          >
            <i
              class="lni lni-zoom-out size-md "
            />
          </a>
        </div>
        <div
          class="canvas-panner"
        />
      </div>
      <div
        class="sidebar  dark-theme"
        id="sidebar"
      >
        <div
          class="content"
        >
          <h3>
            About
          </h3>
          <p>
            Hello, this is JDL-Studio, a tool for drawing
             
            <a
              href="http://www.jhipster.tech/"
              rel="noopener noreferrer"
              target="_blank"
            >
              JHipster
            </a>
             
            JDL diagrams based on the
             
            <a
              href="http://www.jhipster.tech//jdl"
              rel="noopener noreferrer"
              target="_blank"
            >
              JDL syntax
            </a>
            .
             
          </p>
          <p>
            Try and edit the code on the left and watch the diagram change. Any changes are saved to the browser's 
            <i>
              localStorage
            </i>
            , so your diagram should be here the next time, (but no guarantees). If you want persistent storage, sign in with your JHipster Online account to save the JDL.
          </p>
          <p>
            Created by
             
            <a
              href="https://deepu.tech"
              rel="noopener noreferrer"
              target="_blank"
            >
              Deepu KS
            </a>
             
            and hosted on
             
            <a
              href="https://github.com/jhipster/jdl-studio"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
            </a>
            .
          </p>
          <p>
            JDL-Studio was made possible by the cool project
             
            <a
              href="https://github.com/skanaar/nomnoml"
              rel="noopener noreferrer"
              target="_blank"
            >
              nomnoml
            </a>
             
            by
             
            <a
              href="https://github.com/skanaar"
              rel="noopener noreferrer"
              target="_blank"
            >
              Daniel Kallin
            </a>
             
            .
          </p>
          <h2>
            Interaction
          </h2>
          <hr />
          <p>
            The canvas can be panned and zoomed by dragging and scrolling in the right hand third of the canvas.
          </p>
          <img
            alt="interaction tutorial"
            class="invert"
            src="interaction-tutorial.png"
          />
          <p>
            Downloaded image files will be given the filename in the
             
            <kbd>
              #title
            </kbd>
             directive.
          </p>
          <p>
            The editor supports a subset of 
            <i>
              Sublime Text's
            </i>
             
            <a
              href="https://codemirror.net/demo/sublime.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              keymappings
            </a>
            . Hit 
            <code>
              Ctrl-Space
            </code>
             for auto suggestions.
          </p>
          <h2>
            Examples
          </h2>
          <hr />
          <p>
            Refer
             
            <a
              href="http://www.jhipster.tech/jdl"
              rel="noopener noreferrer"
              target="_blank"
            >
              JDL
            </a>
             
            documentation for examples.
          </p>
          <h2>
            Usage
          </h2>
          <hr />
          <p>
            Refer
             
            <a
              href="http://www.jhipster.tech/jdl"
              rel="noopener noreferrer"
              target="_blank"
            >
              JDL
            </a>
             
            documentation for more details.
          </p>
          <label>
            Set font
          </label>
          <code>
            #font: Times
          </code>
          <label>
            Layout the diagram really tight
          </label>
          <code>
            #fontSize: 8
          </code>
          <br />
          <code>
            #spacing: 12
          </code>
          <br />
          <code>
            #padding: 3
          </code>
          <h2>
            Directives
          </h2>
          <hr />
          <p>
            The rendered image style can be customized by setting these directives along with the JDL code. There should not be any empty space before a directive.
          </p>
          <code>
            #arrowSize: 1
          </code>
          <br />
          <code>
            #bendSize: 0.3
          </code>
          <br />
          <code>
            #direction: down | right
          </code>
          <br />
          <code>
            #gutter: 5
          </code>
          <br />
          <code>
            #edgeMargin: 0
          </code>
          <br />
          <code>
            #edges: hard | rounded
          </code>
          <br />
          <code>
            #background: transparent
          </code>
          <br />
          <code>
            #fill: #eee8d5; #fdf6e3
          </code>
          <br />
          <code>
            #fillArrows: false
          </code>
          <br />
          <code>
            #font: Calibri
          </code>
          <br />
          <code>
            #fontSize: 12
          </code>
          <br />
          <code>
            #leading: 1.25
          </code>
          <br />
          <code>
            #lineWidth: 3
          </code>
          <br />
          <code>
            #padding: 8
          </code>
          <br />
          <code>
            #spacing: 40
          </code>
          <br />
          <code>
            #stroke: #33322E
          </code>
          <br />
          <code>
            #title: filename
          </code>
          <br />
          <code>
            #zoom: 1
          </code>
          <br />
          <code>
            #acyclicer: greedy
          </code>
          <br />
          <code>
            #ranker: network-simplex | tight-tree | longest-path
          </code>
          <br />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
